asf_heat_pump_suitability/research/exploratory/sample_buildings_for_labelling.py

```diff
@@ -0,0 +1,401 @@
+    select_las=config["constant"]["sampling_areas"]
+)
+sampling_areas_gdf = sampling_areas_gdf.sjoin(
+    la_boundaries_gdf[["LAD23CD", "LAD23NM", "geometry"]],
+    how="left",
+    predicate="within",
+).drop(columns="index_right")
+sampling_areas_df = pl.from_pandas(sampling_areas_gdf.drop(columns="geometry")).rename(
+    {"ID": "building_id"}
+)
+
+# %%
+# CREATE SAMPLE OF BUILDINGS TO LABEL
+pl.set_random_seed(seed)
+
+# Identify already labelled buildings
+already_labelled = set(sample_df["building_id"].unique()) | set(
+    small_building_sample_df["building_id"].unique()
+)
+already_labelled_df = pl.concat([sample_df, small_building_sample_df])
+
+# Prepare already labelled dataframe with same columns for later sampling
+already_labelled_df = already_labelled_df.with_columns(
+    pl.Series("la_name", ["Plymouth"] * already_labelled_df.height),
+    # Group number of flats
+    pl.when(pl.col("n_flats").is_between(2, 6, closed="both"))
+    .then(pl.lit("2-6_flats"))
+    .when(pl.col("n_flats").is_between(7, 15, closed="both"))
+    .then(pl.lit("7-15_flats"))
+    .when(pl.col("n_flats") > 15)
+    .then(pl.lit("16+_flats"))
+    .otherwise(None)
+    .alias("n_flats_grouped"),
+    # Add labeller
+    pl.Series("labeller", ["reindeer"] * already_labelled_df.height),
+)
+
+# Get buildings containing flats
+buildings_containing_flats = sampling_areas_df.filter(pl.col("property_type_flat"))[
+    "building_id"
+].unique(maintain_order=True)
+
+# Sample buildings containing flats but only if they haven't been labelled
+sample_from_df = (
+    sampling_areas_df.filter(
+        pl.col("building_id").is_in(buildings_containing_flats),
+        ~pl.col("building_id").is_in(already_labelled),
+    )
+    .group_by("building_id")
+    .agg(
+        pl.col("property_type_flat").sum().alias("n_flats"),
+        pl.col("UPRN").count().alias("n_total"),
+        pl.col("LAD23NM").first().alias("la_name"),
+        # Sample from any buildings with >1 flat
+    )
+    .filter(pl.col("n_flats") > 1)
+    .with_columns(
+        # Group number of flats
+        pl.when(pl.col("n_flats").is_between(2, 6, closed="both"))
+        .then(pl.lit("2-6_flats"))
+        .when(pl.col("n_flats").is_between(7, 15, closed="both"))
+        .then(pl.lit("7-15_flats"))
+        .when(pl.col("n_flats") > 15)
+        .then(pl.lit("16+_flats"))
+        .otherwise(None)
+        .alias("n_flats_grouped")
+    )
+)
+
+samples = []
+
+n_flats_groups = ["2-6_flats", "7-15_flats", "16+_flats"]
+
+# Take samples for each Local Authority for each group of flat counts to create sample of 700 buildings
+for la, df in sample_from_df.sort(by="la_name").group_by(
+    "la_name", maintain_order=True
+):
+    # Get the number of samples per LA
+    n = n_samples[la[0]]
+    # Take one third of samples from each group of flat counts
+    for n_flats in n_flats_groups:
+        _n = np.ceil(n / len(n_flats_groups))
```

Comment:
```suggestion
        # proportion of samples from each group of flat counts, in this case it's one third because there are 3 groups
        prop = np.ceil(n / len(n_flats_groups))
```